federation/apis/federation/types.go

```diff
@@ -153,3 +153,16 @@ type ClusterReplicaSetPreferences struct {
 	// A number expressing the preference to put an additional replica to this LocalReplicaSet. 0 by default.
 	Weight int64
 }
+
+// Annotations for a federated service to keep record of service load balancer ingress endpoints
```

Comment:
It seems that this information is nothing else than a cache of the underlying cluster service status.  Can we not use such a cache, rather then duplicate the information, under slightly different names/types here?